### PR TITLE
[POC] Feature Favorite Apps on swipe left

### PR DESCRIPTION
**Pull request by coxtor**
This is more a POC instead of pretty code,  but this PR adds a favorite option when swiping left, including the configurable options.
Issues with this currently:

- No Scrolling, if #442  is merged a whole lot would change
- We should have a central way to query installed apps with icons and a textual description
- There still needs to be a handle that checks whether the return app should be the watch face or the previous menu, however this would currently require a change in every app so I didn't do it to keep this PR more as an idea for now.  [or #478, #480

https://user-images.githubusercontent.com/7712562/125523594-aeaddcfb-67fc-45bf-8246-db844c3d1c76.mp4

]

### CODE REVIEW

**Comment by Avamander** [comment]
Just FYI the commit message is not very accurate, it should describe what the commit actually does.

File: src/displayapp/screens/settings/SettingFavoriteApp.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      private:
+        Controllers::Settings& settingsController;
+        uint8_t optionsTotal;
+        lv_obj_t* cbOption[13];
+        Apps favoriteOptions[13] = {Apps::Music,Apps::Navigation,Apps::StopWatch,Apps::Timer,Apps::None,Apps::FlashLight,Apps::Paint,Apps::Paddle,Apps::Twos,Apps::HeartRate,Apps::Metronome,Apps::Motion,Apps::Steps};   
+        char favoriteNames[13][12] = { "Music","Navigation","Stop Watch","Timer","None","Flash Light","Paint","Paddle","Twos","Heart Rate","Metronome","Motion","Steps"};
+      };
+    }
+  }
+}
```
Comment (Avamander) [inline src/displayapp/screens/settings/SettingFavoriteApp.h:31]: This file hasn't been formatted with clang-format I think

**Review by Avamander — CHANGES_REQUESTED** [review]
On a first glance this looks nice, but there's a few inconsistencies.

**Comment by Riksu9000** [comment]
>We should have a central way to query installed apps with icons and a textual description

I've mentioned something similar to this in #462. It should be easy to combine the [separate arrays in ApplicationList.cpp](https://github.com/JF002/InfiniTime/blob/57b339707861c5688f5d432f1506a99df6bb0fce/src/displayapp/screens/ApplicationList.cpp#L48) and move them to [ApplicationList.h](https://github.com/JF002/InfiniTime/blob/develop/src/displayapp/screens/ApplicationList.h), where it could be easily queried, without the description though.

>There still needs to be a handle that checks whether the return app should be the watch face or the previous menu

I think this can be done by creating separate entries in the [LoadApp function in DisplayApp.cpp](https://github.com/JF002/InfiniTime/blob/57b339707861c5688f5d432f1506a99df6bb0fce/src/displayapp/DisplayApp.cpp#L259) for when the app is opened from watchface and for when it is opened from settings and they'd have different ReturnApps.

**Comment by Riksu9000** [comment]
Potentially if #492 gets merged, the favorite app could be selected by long pressing on an app in the app list. The selected favorite app could be a different color, or have a small star.

>>There still needs to be a handle that checks whether the return app should be the watch face or the previous menu

>I think this can be done by creating separate entries in the LoadApp function in DisplayApp.cpp

I was thinking of just the settings menu. I realise now that this would also be required for all apps to know whether to return to the app list or watch face, which obviously isn't a great solution. However could `  ReturnApp(Apps::Clock, ...);
` simply be run after this line?
https://github.com/JF002/InfiniTime/blob/3e0974eacdad15a47c2f359cf7f02304806f5513/src/displayapp/DisplayApp.cpp#L192

**Comment by Izaic** [comment]
This would be a really great feature to have for Version 1.4

**Comment by coxtor** [comment]
Hi all, I'm not quite sure what went wrong that this PR has so many commits - however I got to implement the "long tap to select favorite" option. However I need help here - for some reason the watch crashes when I don't use a static Application when saving a favorite (tile.cpp -> apps[buttonId] ) . Can someone help me out here?

**Comment by kieranc** [comment]
Any chance you can clean up the PR so we can have a look more easily? I tested this yesterday and it seems to work well but it's a bit of a pain to merge currently

**Comment by kieranc** [comment]
I hope you'll bring this back, I was testing it because people were asking for it, and it works nicely!

**Comment by coxtor** [comment]
I did some cleaning up and we can continue the discussion on PR #708

**Pull request closed without merging**